src/cascadia/TerminalSettingsModel/TerminalSettings.h

```diff
@@ -0,0 +1,221 @@
+        GETSET_SETTING(Model::TerminalSettings, Windows::Foundation::IReference<uint32_t>, StartingTabColor, nullptr);
+
+        // ------------------------ End of Core Settings -----------------------
+
+        GETSET_SETTING(Model::TerminalSettings, hstring, ProfileName);
+        GETSET_SETTING(Model::TerminalSettings, bool, UseAcrylic, false);
+//    public: /* Returns true if the user explicitly set the value, false otherwise*/
+//        bool HasUseAcrylic() const
+//        {
+//            return _UseAcrylic.has_value();
+//        }
+//
+//        Model::TerminalSettings UseAcrylicOverrideSource()
+//        { /*user set value was not set*/ /*iterate through parents to find one with a value*/
+//            for (auto& parent : _parents)
+//            {
+//                if (auto source{ parent->_getUseAcrylicOverrideSourceImpl() })
+//                {
+//                    return source;
+//                }
+//            }
+//
+//            /*no value was found*/
+//            return nullptr;
+//        }
+//
+//        /* Returns the resolved value for this setting */ /* fallback: user set value --> inherited value --> system set value */
+//        bool UseAcrylic() const
+//        {
+//            const auto val{ _getUseAcrylicImpl() };
+//            return val ? *val : bool{ false };
+//        }
+//
+//        /* Overwrite the user set value */
+//        void UseAcrylic(const bool& value)
+//        {
+//            _UseAcrylic = value;
+//        }
+//
+//        /* Clear the user set value */
+//        void ClearUseAcrylic()
+//        {
+//            _UseAcrylic = std::nullopt;
+//        }
+//
+//    private:
+//        std::optional<bool> _UseAcrylic{ std::nullopt };
+//        std::optional<bool> _getUseAcrylicImpl() const
+//        { /*return user set value*/
+//            if (_UseAcrylic)
+//            {
+//                return _UseAcrylic;
+//            }
+//
+//            /*user set value was not set*/ /*iterate through parents to find a value*/
+//            for (auto parent : _parents)
+//            {
+//                if (auto val{ parent->_getUseAcrylicImpl() })
+//                {
+//                    return val;
+//                }
+//            }
+//
+//            /*no value was found*/
+//            return std::nullopt;
+//        }
+//        Model::TerminalSettings _getUseAcrylicOverrideSourceImpl() const
+//        { /*we have a value*/
+//            if (_UseAcrylic)
+//            {
+//                return *this;
+//            }
+//
+//            /*user set value was not set*/ /*iterate through parents to find one with a value*/
+//            for (auto& parent : _parents)
+//            {
+//                if (auto source{ parent->_getUseAcrylicOverrideSourceImpl() })
+//                {
+//                    return source;
+//                }
+//            }
+//
+//            /*no value was found*/
+//            return nullptr;
+//        }

```

Comment:
Ok, this sucks, and is the main reason this PR is in draft right now. The macro has a bug in it that is only repro-able if we don't expand the macro (I hate it). On a settings reload, we get a crash :(

Call stack:
![call stack](https://user-images.githubusercontent.com/11050425/109543108-cbe03780-7a7a-11eb-9120-eedd5f902bcc.png)

Error:
```
**std::_Atomic_address_as<long,std::_Atomic_padded<unsigned int> >**(...) returned 0xFFFFFFFFFFFFFFFF.
```

Surprisingly, this is not an issue on main. So it seems to be an issue with ref counting the com_ptr?

@DHowett mind taking a look here? It's pretty bizarre.

Reply:
Discussed offline. The issue was that I was using `attach` instead of `copy_from`